core/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/util/SchemaVersionUtilsTest.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.core.util
+
+import com.amazon.opendistroforelasticsearch.alerting.core.model.XContentTestBase
+import org.elasticsearch.cluster.metadata.IndexMetaData
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class SchemaVersionUtilsTest : XContentTestBase {
+
+    @Test
+    fun `test get schema version`() {
+        val message = "{\"user\":{ \"name\":\"test\"},\"_meta\":{\"schema_version\": 1}}"
+
+        val schemaVersion = SchemaVersionUtils.getSchemaVersion(message)
+        assertEquals(1, schemaVersion)
+    }
+
+    @Test
+    fun `test get schema version without _meta`() {
+        val message = "{\"user\":{ \"name\":\"test\"}}"
+
+        val schemaVersion = SchemaVersionUtils.getSchemaVersion(message)
+        assertEquals(0, schemaVersion)
+    }
+
+    @Test
+    fun `test get schema version without schema_version`() {
+        val message = "{\"user\":{ \"name\":\"test\"},\"_meta\":{\"test\": 1}}"
+
+        val schemaVersion = SchemaVersionUtils.getSchemaVersion(message)
+        assertEquals(0, schemaVersion)
```

Comment:
We should have a test for invalid / negative schema versions.

Reply:
Will add it